django/urls/converters.py

```diff
@@ -0,0 +1,48 @@
+from django.conf import settings
+from django.utils import lru_cache
+
+
+class BaseConverter(object):
+    pass
+
+
+class IntConverter(BaseConverter):
+    regex = '[0-9]+'
+
+    def to_python(self, value):
+        return int(value)
+
+    def to_url(self, value):
+        return str(value)
+
+
+class StringConverter(BaseConverter):
+    regex = '[^/]+'
+
+    def to_python(self, value):
+        return value
+
+    def to_url(self, value):
+        return value
+
+
+DEFAULT_CONVERTERS = {
```

Comment:
I'm against providing `float` (see my email to django-developers.

The regex for UUID must enforce the location of dashes to avoid the "duplicate URLs for the same view " problem.